vignettes/articles/05-data-cleaning.Rmd

```diff
@@ -210,31 +209,29 @@ If you want to summarize at a finer/coarser scale, you can also create other gro
 The simplest alternative is a rounded timestamp variable; for example by using `mutate(ts.h = plyr::round_any(ts, 3600)`.
 Other options are to just use date (e.g `date = as_date(ts)`).
 
-Here we create a summary data frame that we can filter to produce different plots
+Here we create a summary data frame that we can filter to produce different plots:
 
 ```{r eval = TRUE}
-df.plot <- tbl.alltags.sub %>%
+df.summary <- tbl.alltags.sub %>%
   filter(tagProjID == 176, # keep only tags registered to the sample project
          !is.na(recvDeployLat) | !(recvDeployLat == 0)) %>% # drops data without lon/lat
   group_by(motusTagID, runID, recvDeployName, ambigID, 
            tagDepLon, tagDepLat, recvDeployLat, recvDeployLon) %>%
   #summarizing by runID to get max run length and mean time stamp:
-  summarize(max.runLen = max(runLen, na.rm = TRUE), 
-            ts = mean(ts, na.rm = TRUE), .groups = "drop") %>% 
+  summarize(max.runLen = max(runLen), 
+            ts = mean(ts)) %>% 
   arrange(motusTagID, ts) %>%
-  collect() %>%
-  mutate(ts = plyr::round_any(ts, 3600),
-         ts = as_datetime(ts, tz = "UTC", origin = "1970-01-01"))
```

Comment:
I'm a bit confused by you ran into errors with the timestamp. Any errors should have prevented the article from compiling, but originally it did. I think all the time stamp issues you ran into were because these two lines were omitted early on.

Reply:
Sorry, I should have given a more detailed explanation. This didn't cause an error, but summarized the data differently than the old R Book resulting in slightly different plots. If I'm interpreting it correctly, the text describes a method to summarize by `runID`, then offers alternatives such as a rounded timestamp variable (`ts.h = plyr::round_any(ts, 3600)`). In the old R Book, it doesn't actually implement this timestamp rounded to hour, it just calculates a `mean(ts)` for each `runID`, but confusingly calls that `ts.h`. I modified it to summarize by `runID`, but just keeping `mean(ts)` as `ts`, not `ts.h`. So I do think we shouldn't include `ts = plyr::round_any(ts, 3600)` because it alters the plot (see tag 16036), but I think you're right that some of the later timestamp conversions could be avoided if we kept the last line (`ts = as_datetime(ts, tz = "UTC", origin = "1970-01-01")`). So I think this is the code chunk we want:

```
df.summary <- tbl.alltags.sub %>%
  filter(tagProjID == 176, # keep only tags registered to the sample project
         !is.na(recvDeployLat) | !(recvDeployLat == 0)) %>% # drops data without lon/lat
  group_by(motusTagID, runID, recvDeployName, ambigID, 
           tagDepLon, tagDepLat, recvDeployLat, recvDeployLon) %>%
  #summarizing by runID to get max run length and mean time stamp:
  summarize(max.runLen = max(runLen), 
            ts = mean(ts)) %>% 
  arrange(motusTagID, ts) %>%
  collect() %>% 
  mutate(ts = as_datetime(ts, tz = "UTC", origin = "1970-01-01"))
```